docs/reference/glossary.md

```diff
@@ -0,0 +1,161 @@
+
+## Sector
+
+Storage miners store data on behalf of the Filecoin network in fixed-size blocks of data called _sectors_. Currently, only 32GiB and 64GiB sectors are supported.
+
+## Slash
+
+When a [fault](#fault) is registered for a [sector](#sector), the Filecoin network will _slash_ the [storage miner](#storage-miner) that is supposed to be storing the sector; that is, it will assess penalties to the miner (to be paid out of the [collateral](#collateral) fronted by te miner) for their failure to uphold their pledge of storage. When slashing takes place, the power a miner earns for the associated sector is subtracted from the miner's total power for the purposes of [election](#election).
+
+## Storage miner
+
+A _storage miner_ is a Filecoin participant that stores data on behalf of the network. Storage miners are rewarded for this service through payments by clients that contract their services, as well as by periodic authorization to extend the Filecoin [blockchain](#blockchain) with [blocks](#block) of their own creation. When they create a block, storage miners are rewarded with newly minted filecoin, and by the transaction fees they can levy on other participants seeking to include [messages](#message) in the block.
+
+## Storage power
+
+A [storage miner's](#storage-miner) _storage power_ is a value roughly proportional to the amount of storage capacity they make available on behalf of the network via [capacity commitments](#capacity-commitment) or [storage deals](#deal). Storage power is used to select storage miners for rewards in proportion to their contributions to the total network storage capacity.
+
+## Zero-knowledge succinct non-interactive argument of knowledge (zk-SNARK)
+
+An _argument of knowledge_ is a construction by which one party, called the _prover_, can convince another, the _verifier_, that the prover has access to some piece of information. There are several possible constraints on such constructions:
+
+- A _non-interactive_ argument of knowledge has the requirement that just a single message, sent from the prover to the verifier, should serve as a sufficient argument.
+
+- A _zero-knowledge_ argument of knowledge has the requirement that the verifier should not need access to the knowledge the prover has access to in order to verify the prover's claim.
+
+- A _succinct_ argument of knowledge is one that can be "quickly" verified, and which is "small", for appropriate definitions of both of those terms.
+
+A zero-knowledge, succinct non-interactive argument of knowledge (zk-SNARK) embodies all of these properties. Filecoin utilizes these constructions to enable its distributed network to efficiently verify that [storage miners](#storage-miner) are storing files they pledged to store, without requiring the verifiers to maintain copies of these files themselves.
+
+## Testnet
```

Comment:
Perhaps link to https://docs.filecoin.io/how-to/networks/#testnet somewhere in this definition.

Reply:
Good call! Made the definition more concise and deferred to that page.